### DISCUSSION TITLE
[Help] How to allow players to only use /player JUST_MYSELF shadow? Is this even possible?

### DISCUSSION THREAD

**Discussion by RandomGgames** [gnembon/fabric-carpet#1317] [Q&A] [answered]
I run a survival server, however I only want to allow users to have access to shadow, just for themselves. I've figured out that I can use PlayerRoles to allow "player.*", however that allows complete access to the command. I have tried a couple things but can't seem to get something like "player.@s.shadow" to work. 

Is there a way to do this?

**Comment by replaceitem** [comment, marked as answer]
The best way would be to create a Scarpet script to wrap the `/player shadow` command.

Heres a quick example:

```py
__command() -> (
  run('player ' + player()~'command_name' + ' shadow');
)
```

If you save this in the `scripts` folder inside the world folder as a file called `shadow.sc`, then you can load the script with `/script load shadow` and all players can use `/shadow`, which will shadow them.

**Reply by RandomGgames** [reply]
This works great! Does exactly what I need. One issue I've noticed is that it doesn't seem to work after I restart the server...

**Reply by RandomGgames** [reply]
Sometimes it sends "Tried to mask vanilla command." which might be an error...?

**Reply by replaceitem** [reply]
In the newest carpet version, it should autoload on start. If not, make sure the `scriptsAutoload` is enabled: `/carpet setDefault scriptsAutoload true`
Otherwise, try to add this config line to the top of the script:
```
__config()->{'stay_loaded'->true};
```

The *Tried to mask vanilla command.* means that it tried to add a command that already exists, if you are using `shadow.sc` that shouldn't be the case, but i think that sometimes also appears unintentionally when reloading the script.
